docs/topics/opt-in-requirements.md

```diff
@@ -271,21 +393,56 @@ fun getTime(): Time {}
+
+## Require opt-in to extend API
+
+There may be times when you want more granular control over which specific parts of your API can be used and
+extended. For example, when you have some API that is stable to use but:
+
+* **Unstable to implement** due to ongoing evolution, such as when you have a family of interfaces where you expect to add new abstract functions without default implementations.
+* **Delicate or fragile to implement**, such as individual functions that need to behave in a coordinated manner.
+* **Has a contract that may be weakened in the future** in a backwards-incompatible manner for external implementations, such as changing an input parameter `T` to a nullable version `T?` where the code didn't previously consider `null` values.
+
+In such cases, you want to require users to opt in to your API before they can extend it further.
+For example, by inheriting from the API or declaring implementations for abstract functions. Using the [`@SubclassOptInRequired`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-subclass-opt-in-required/)
+annotation, you can enforce this behavior for [open](inheritance.md) or [abstract classes](classes.md#abstract-classes) and [non-functional interfaces](interfaces.md).
+
+To add the opt-in requirement to an API element, use the [`@SubclassOptInRequired`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-subclass-opt-in-required/)
+annotation with a reference to the annotation class:
+
+```kotlin
+@RequiresOptIn(
+ level = RequiresOptIn.Level.WARNING,
+ message = "Interfaces in this library are experimental"
+)
+annotation class UnstableApi()
+
+@SubclassOptInRequired(UnstableApi::class)
+// An interface requiring opt-in to extend
+interface CoreLibraryApi 
+```
+
+Note that when you use the `@SubclassOptInRequired` annotation to require opt-in, the requirement is not propagated to
+any [inner or nested classes](nested-classes.md).
+
+For a real-world example of how to use the `@SubclassOptInRequired` annotation in your API, check out the
+[`SharedFlow`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-shared-flow/)
+interface in the `kotlinx.coroutines` library.
 
 ## Opt-in requirements for pre-stable APIs
 
 If you use opt-in requirements for features that are not stable yet, carefully handle the API graduation to avoid 
-breaking the client code.
+breaking client code.
 
-Once your pre-stable API graduates and is released in a stable state, remove its opt-in requirement annotations from declarations.
-The clients will be able to use them without restriction. However, you should leave the annotation classes in modules so that 
-the existing client code remains compatible.
+Once your pre-stable API graduates and is released in a stable state, remove any opt-in requirement annotations from 
+your declarations. The clients can then use them without restriction. However, you should leave the annotation classes 
```

Comment:
```suggestion
your declarations. This allows clients to use them without restrictions. However, you should leave the annotation classes 
```

Reply:
Do you think we should keep in "should" here?
I think it might be better to remove it to ensure compatibility.

Reply:
@BlondeHex do you mind if we remove "should" here? Or should we leave it like this because it's intended as a recommendation and not a hard rule?